### PR TITLE
Catch InputCoercionException thrown by Jackson parser

### LINKED ISSUE #57272: ignored_malfored flag is ignored in numeric types for out of range values
It can be reproduce by:

```
PUT test
{
    "mappings": {
        "properties": {
            "number": {
                "type": "long",
                "ignore_malformed": true
            }
        }
    }
}

PUT test/_doc/1
{
    "number": 9223372036854775808
}
```

This will throw an error in 7.7.0:

```
{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "failed to parse field [number] of type [long] in document with id '1'. Preview of field's value: '9223372036854775808'"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "failed to parse field [number] of type [long] in document with id '1'. Preview of field's value: '9223372036854775808'",
    "caused_by": {
      "type": "input_coercion_exception",
      "reason": "Numeric value (9223372036854775808) out of range of long (-9223372036854775808 - 9223372036854775807)\n at [Source: (org.elasticsearch.common.bytes.AbstractBytesReference$MarkSupportingStreamInputWrapper); line: 2, column: 34]"
    }
  }
```

### PR DESCRIPTION
**Pull request by iverase**
Jackson 2.10 library has added a new type of error that is thrown when a numeric value is out of range. This error should be catched and handle properly in case the flag `ignore_malformed` has been set to true.

fixes #57272

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/index/mapper/LongIndexingDocTests.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ESSingleNodeTestCase;
+
+public class LongIndexingDocTests extends ESSingleNodeTestCase {
```
Comment (cbuescher) [inline server/src/test/java/org/elasticsearch/index/mapper/LongIndexingDocTests.java:29]: nit: Since this is testing NumberFieldMapper behaviour, would it make sense to put this test case in `NumberFieldMapperTests` instead? We wouldn't need a new isolated test class with all the setup/teardown overhead for it in that case.

**Review by cbuescher — APPROVED** [review]
LGTM, left a small question around potentially moving the test elsewhere, but feel free to decide.

File: server/src/test/java/org/elasticsearch/index/mapper/LongIndexingDocTests.java (hunk 1 above)
Comment (iverase) [reply server/src/test/java/org/elasticsearch/index/mapper/LongIndexingDocTests.java:29]: I can only reproduce if the test is of the type `ESSingleNodeTestCase` which is not the case for `NumberFieldMapperTests`

Comment (cbuescher) [reply server/src/test/java/org/elasticsearch/index/mapper/LongIndexingDocTests.java:29]: Strange, from my IDE it looked like it extends "ESSingleNodeTestCase". Let me try again...

Comment (cbuescher) [reply server/src/test/java/org/elasticsearch/index/mapper/LongIndexingDocTests.java:29]: Works for me :scratcheshead: I moved your test over and only had to change the index name since the "test" index seems to be in use elsewhere already. Failed when removing the new exception type from the catch statement, passes otherwise.

Comment (iverase) [reply server/src/test/java/org/elasticsearch/index/mapper/LongIndexingDocTests.java]: You are totally right, I have moved the test so we don't add a new whole class. Thanks @cbuescher!
